### DISCUSSION TITLE
Bug report: Commit comments don't seem to be XSS sanitized

### DISCUSSION THREAD

**Discussion by ShadyMedic** [community/community#27726] [Pull Requests] [answered]
Hey everyone.

So I posted [this comment](https://github.com/Team-VoW/VoicesOfWynn-Website/commit/973ce0f35f8d7251184ea69b4e94f833f9a6db23#r79827001) on my commit a few minutes ago. The contents of the comment were:
![Comment textarea](https://user-images.githubusercontent.com/19894267/181934027-43bfe4a4-004a-45f9-9703-f7b502f6b43c.png)

But when I posted the comment, this appeared:
![Displayed comment](https://user-images.githubusercontent.com/19894267/181934067-3bbe9e1f-5316-46f2-888f-b6bbdf402946.png)

It seems like the `<div>` text was evaluated as HTML, which means that the contents of the comment weren't sanitized against a XSS attack.

Luckily it doesn't seem to evaluate `<script>` tags, but it still seems like quite a big issue.
![<script> not working](https://user-images.githubusercontent.com/19894267/181934220-252f3e45-f166-4b65-a52a-13ed0d3e5ccb.png)

**Comment by airtower-luna** [comment, marked as answer]
That does look problematic, yes. :open_mouth: You might want to also send a [but report directly to support](https://support.github.com/contact/bug-report). I know there are GitHub staff around here, but a ticket might be a little faster than waiting for one of them to notice this discussion.

**Reply by ShadyMedic** [reply]
Thanks, I couldn't find that link. I submitted it as a security alert.

**Comment by spenserblack** [comment]
[GitHub flavored markdown supports HTML](https://github.github.com/gfm/#html-blocks)

Some frequently used tags are `<summary>` and `<details>`, for example, to shorten a comment but still provide verbose details for those interested.
<details>
<summary>Example</summary>
You viewed the details!
</details>

HTML comments, `<!-- comment -->`, are also frequently used. You'll see them in many issue templates.

I wish the allowed HTML blocks were better documented, but it looks like this is the current allowlist.
https://github.com/gjtorikian/html-pipeline/blob/03ae30d713199c2562951d627b98b75dc16939e4/lib/html/pipeline/sanitization_filter.rb#L42-L109

(https://github.blog/2012-11-27-html-pipeline-chainable-content-filters/)

**Reply by airtower-luna** [reply]
Thank you for pointing that out! I was aware *some* HTML elements are supported, but not that it was that many. So the system is working as designed, though now I have my doubts about the design. :sweat_smile:
